apps/system/test/marionette/lockscreen_camera_button_test.js

```diff
@@ -0,0 +1,148 @@
+        var theCameraApp = client.executeScript(function() {
+          return window.wrappedJSObject.document
+            .querySelector('.secureAppWindow.fullscreen-app.active.render');
+        });
+        if (theCameraApp) {
+          return theCameraApp.displayed();
+        } else {
+          return false;
+        }
+      });
+    })
+    .then(done)
+    .catch(done);
+  });
+
+  test('after launch the app, press power button twice should dismiss it',
+  function(done) {
+    client.switchToFrame();
+    new Promise(function(resolve) {
+      lockScreen.slideToUnlock(resolve);
+    })
+    .then(function() {
+      var notShownYet = client.executeScript(function() {
+        return null === window.wrappedJSObject.document
+          .querySelector('.secureAppWindow.fullscreen-app.active.render');
+      });
+      assert.ok(notShownYet);
+      client.findElement('#lockscreen-alt-camera-button').tap();
+      client.waitFor(function() {
+        var theCameraApp = client.executeScript(function() {
+          return window.wrappedJSObject.document
+            .querySelector('.secureAppWindow.fullscreen-app.active.render');
+        });
+        if (theCameraApp) {
+          return theCameraApp.displayed();
+        } else {
+          return false;
+        }
+      });
+    })
+    .then(function() {
+      client.executeScript(function() {
+        window.wrappedJSObject.Service.request('turnScreenOff', true);
+        window.wrappedJSObject.Service.request('turnScreenOn', true);
+      });
+      client.waitFor(function() {
+        var dismissed = client.executeScript(function() {
+          return null === window.wrappedJSObject.document
+            .querySelector('.secureAppWindow.fullscreen-app.active.render');
+        });
+        return dismissed;
+      });
+      assert.ok(!client.findElement('#lockscreen-passcode-code').displayed());
+    })
+    .then(done)
+    .catch(done);
+  });
+
+  test('after launch the app, press home button should dismiss it',
+  function(done) {
+    client.switchToFrame();
+    new Promise(function(resolve) {
+      lockScreen.slideToUnlock(resolve);
+    })
+    .then(function() {
+      var notShownYet = client.executeScript(function() {
+        return null === window.wrappedJSObject.document
+          .querySelector('.secureAppWindow.fullscreen-app.active.render');
+      });
+      assert.ok(notShownYet);
+      client.findElement('#lockscreen-alt-camera-button').tap();
+      client.waitFor(function() {
+        var theCameraApp = client.executeScript(function() {
+          return window.wrappedJSObject.document
+            .querySelector('.secureAppWindow.fullscreen-app.active.render');
+        });
+        if (theCameraApp) {
+          return theCameraApp.displayed();
+        } else {
+          return false;
+        }
+      });
+    })
+    .then(function() {
+      client.executeScript(function() {
+        window.wrappedJSObject.dispatchEvent(new CustomEvent('home'));
```

Comment:
I am not entire sure if there is better way to similar this (send mocked key event from Gecko?), but we should do that if there is ...